xrdsst/controllers/user.py

```diff
@@ -0,0 +1,69 @@
+import subprocess
+
+from cement import ex
+from xrdsst.controllers.base import BaseController
+from xrdsst.core.util import get_admin_credentials, get_ssh_key, get_ssh_user
+from xrdsst.resources.texts import texts
+
+
+class UserController(BaseController):
+    _GROUP_NAMES = ['xroad-security-officer',
+                    'xroad-registration-officer',
+                    'xroad-service-administrator',
+                    'xroad-system-administrator',
+                    'xroad-securityserver-observer']
+
+    class Meta:
+        label = 'user'
+        stacked_on = 'base'
+        stacked_type = 'nested'
+        description = texts['user.controller.description']
+
+    @ex(help="Create admin user", arguments=[])
+    def create_admin(self):
+        self.create_user(self.load_config())
+
+    def create_user(self, conf):
+        for security_server in conf["security_server"]:
+            self.log_debug('Creating admin user for security server: ' + security_server['name'])
```

Comment:
Here, only attempt user creation when administrator credentials are present, currently it then fails ungracefully:

```
$ xrdsst -c config/worknewss5.yaml user create-admin 
Traceback (most recent call last):
  File "/home/uns/dev/niis/xrdsst-review/X-Road-Security-Server-toolkit/env/bin/xrdsst", line 11, in <module>
    load_entry_point('xrdsst', 'console_scripts', 'xrdsst')()
  File "/home/uns/dev/niis/xrdsst-review/X-Road-Security-Server-toolkit/xrdsst/main.py", line 237, in main
    app.run()
  File "/home/uns/dev/niis/xrdsst-review/X-Road-Security-Server-toolkit/env/lib/python3.8/site-packages/cement/core/foundation.py", line 916, in run
    return_val = self.controller._dispatch()
  File "/home/uns/dev/niis/xrdsst-review/X-Road-Security-Server-toolkit/env/lib/python3.8/site-packages/cement/ext/ext_argparse.py", line 808, in _dispatch
    return func()
  File "/home/uns/dev/niis/xrdsst-review/X-Road-Security-Server-toolkit/xrdsst/controllers/user.py", line 24, in create_admin
    self.create_user(self.load_config())
  File "/home/uns/dev/niis/xrdsst-review/X-Road-Security-Server-toolkit/xrdsst/controllers/user.py", line 29, in create_user
    admin_credentials = get_admin_credentials(security_server, conf)
  File "/home/uns/dev/niis/xrdsst-review/X-Road-Security-Server-toolkit/xrdsst/core/util.py", line 8, in get_admin_credentials
    admin_credentials = security_server["admin_credentials"] if security_server.get("admin_credentials", "") else config["admin_credentials"]
KeyError: 'admin_credentials'
```

as it is outside operation graph and has no required/validated inputs defined.

Reply:
Will now fail with exception if admin_credentials, ssh_key or ssh_user is missing as these are required parameters